### PR TITLE
feat: Apps-1558 create alphabetical browse by

### PR DESCRIPTION
**Pull request by matheusgomes062**
**Component Changed:** createAlphabeticalBrowseBy.vue from #383

**Stories:** ~/stories/CreateAlphabeticalBrowseBy.stories.stories.js

**Notes:**

**Checklist:**

-   [x] I double-checked it looks like the designs
-   [x] I completed any required mobile breakpoint styling
-   [x] I completed any required hover state styling
-   [x] I included a working Storybook file
-   [x] I included a Story that showed some edge cases working correctly (long text, short screen, missing image, etc.)
-   [x] I assigned this PR to someone to review

### CODE REVIEW

File: src/styles/variables-scss.scss [hunk 1]
```diff
@@ -10,7 +10,7 @@
 // $lte-phone-landscape: "only screen and (max-width: 750px) and (orientation: landscape)";
 // $has-hover: "(hover: hover)";
 @import "~ucla-library-design-tokens/scss/variables.scss";
-// @import "~ucla-library-design-tokens/scss/typography.scss";
+@import "~ucla-library-design-tokens/scss/typography.scss";
```
Comment (pghorpade) [inline src/styles/variables-scss.scss:13]: we need to keep this commented like before.

Comment (pghorpade) [reply src/styles/variables-scss.scss:13]: Also do not commit your package-lock.json, once you have the linux machine, you can try npm install again.

File: src/stories/AlphabeticalBrowseBy.stories.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export default {
+    title: "Search / AlphabeticalBrowseBy",
+    component: AlphabeticalBrowseBy,
+}
+
+export const Default = () => ({
+    components: { AlphabeticalBrowseBy },
+    template: `<alphabetical-browse-by/>`,
+})
```
Comment (pghorpade) [inline src/stories/AlphabeticalBrowseBy.stories.js:11]: We need another story to simulate the click action, there is storybook addon for it.

File: src/lib-components/AlphabeticalBrowseBy.vue [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        }
+    },
+    props: {},
+    computed: {
+        alphabet() {
+            return this.alphabetFirstHalf.concat(this.alphabetSecondHalf)
+        },
+    },
+    created() {
+        window.addEventListener("resize", this.getSize)
```
Comment (pghorpade) [inline src/lib-components/AlphabeticalBrowseBy.vue:87]: the main library website nuxt repo has vuex to do this, will go over this with you tomorrow.

File: src/lib-components/AlphabeticalBrowseBy.vue [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                "X",
+                "Y",
+                "Z",
+            ],
+            isTablet: false,
+            vw: window.innerWidth,
+            vh: window.innerHeight,
+        }
+    },
+    props: {},
```
Comment (pghorpade) [inline src/lib-components/AlphabeticalBrowseBy.vue:80]: we need a selectedLetter prop, if user bookmarks a page with selected letter

**Review by pghorpade — CHANGES_REQUESTED** [review]

File: src/styles/variables-scss.scss (hunk 1 above)
Comment (matheusgomes062) [reply src/styles/variables-scss.scss:13]: But I'm using somethings from typography like @include step-1 and step-2

**Review by pghorpade — APPROVED** [review]
👍

File: src/lib-components/AlphabeticalBrowseBy.vue [hunk 5]
```diff
@@ -111,25 +111,21 @@ export default {
     computed: {
         parsedAlphabet: {
             get() {
-                let letterClass = "letter "
+                let letterClass = "letter"
```
Comment (matheusgomes062) [inline src/lib-components/AlphabeticalBrowseBy.vue:114]: I don't think this line is being used, since we have the same line inside the return.

**Review by pghorpade — APPROVED** [review]
👍

**Pull request merged**